### PR TITLE
Add return types - batch 1/n

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | #40154
| License       | MIT
| Doc PR        | -

Extracted from #42496 for components that shouldn't be extended very often.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:green_book: (psalm failure are false-positives here.)

File: src/Symfony/Component/Cache/Adapter/PhpArrayAdapter.php [hunk 1]
```diff
@@ -68,10 +68,8 @@ static function ($key, $value, $isHit) {
      *
      * @param string                 $file         The PHP file were values are cached
      * @param CacheItemPoolInterface $fallbackPool A pool to fallback on when an item is not hit
-     *
-     * @return CacheItemPoolInterface
      */
-    public static function create(string $file, CacheItemPoolInterface $fallbackPool)
+    public static function create(string $file, CacheItemPoolInterface $fallbackPool): CacheItemPoolInterface
```
Comment (derrabus) [inline src/Symfony/Component/Cache/Adapter/PhpArrayAdapter.php:72]:
```suggestion
    public static function create(string $file, CacheItemPoolInterface $fallbackPool): static
```

I don't quite understand why the previously annotated return type was so wide, tbh.

File: src/Symfony/Component/Stopwatch/Section.php [hunk 2]
```diff
@@ -47,7 +47,7 @@ public function __construct(float $origin = null, bool $morePrecision = false)
      *
      * @return self|null The child section or null when none found
      */
-    public function get(string $id)
+    public function get(string $id): ?\Symfony\Component\Stopwatch\Section
```
Comment (derrabus) [inline src/Symfony/Component/Stopwatch/Section.php]:
```suggestion
    public function get(string $id): ?self
```

File: src/Symfony/Component/Stopwatch/Section.php [hunk 3]
```diff
@@ -65,7 +65,7 @@ public function get(string $id)
      *
      * @return self
      */
-    public function open(?string $id)
+    public function open(?string $id): \Symfony\Component\Stopwatch\Section
```
Comment (derrabus) [inline src/Symfony/Component/Stopwatch/Section.php]:
```suggestion
    public function open(?string $id): self
```

**Review by derrabus — APPROVED** [review]

**Pull request merged**